### ISSUE TITLE
APPCRASH in sshd.exe

### ISSUE THREAD

**Issue by sddim** [PowerShell/Win32-OpenSSH#1113]
**"OpenSSH for Windows" version**
v7.6.0.0p1-Beta
 
**Server OperatingSystem**
Windows Server 2012 R2 Datacenter

**Client OperatingSystem**
Windows Server 2012 R2 Datacenter

**What is failing**
Login with local user fails, connection is closed immediately after successful verification of key, seemingly due to a crash by a spawned sshd.exe .
EDIT: This crash happens reliably with one specific local user, but doesn't happen with one different specific user. Both users are members of the same groups. 

**Actual output**

APPCRASH WER:

Version=1
EventType=APPCRASH
EventTime=131665361022690972
ReportType=2
Consent=1
ReportIdentifier=fea08b81-30e5-11e8-80f7-66fbb5b1f554
IntegratorReportIdentifier=fea08b80-30e5-11e8-80f7-66fbb5b1f554
NsAppName=sshd.exe
Response.type=4
Sig[0].Name=Anwendungsname
Sig[0].Value=sshd.exe
Sig[1].Name=Anwendungsversion
Sig[1].Value=7.6.0.0
Sig[2].Name=Anwendungszeitstempel
Sig[2].Value=5a972a52
Sig[3].Name=Fehlermodulname
Sig[3].Value=sshd.exe
Sig[4].Name=Fehlermodulversion
Sig[4].Value=7.6.0.0
Sig[5].Name=Fehlermodulzeitstempel
Sig[5].Value=5a972a52
Sig[6].Name=Ausnahmecode
Sig[6].Value=c0000005
Sig[7].Name=Ausnahmeoffset
Sig[7].Value=0000000000057eea
DynamicSig[1].Name=Betriebsystemversion
DynamicSig[1].Value=6.3.9600.2.0.0.400.8
DynamicSig[2].Name=Gebietsschema-ID
DynamicSig[2].Value=1031
UI[2]=C:\Program Files\openSSH\sshd.exe
FriendlyEventName=Nicht mehr funktionsfähig
ConsentKey=APPCRASH
AppName=OpenSSH for Windows
AppPath=C:\Program Files\openSSH\sshd.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=A0B5F38D2EC9B3A3F58F96598325B3D5

Daemon log:
29708 2018-03-26 12:18:09.099 debug3: mm_answer_keyallowed: key is allowed
29708 2018-03-26 12:18:09.099 debug3: mm_request_send entering: type 23
29708 2018-03-26 12:18:09.099 debug3: mm_sshkey_verify entering [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_request_send entering: type 24 [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_sshkey_verify: waiting for MONITOR_ANS_KEYVERIFY [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_request_receive_expect entering: type 25 [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_request_receive entering [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_request_receive entering
29708 2018-03-26 12:18:09.099 debug3: monitor_read: checking request 24
29708 2018-03-26 12:18:09.099 debug3: mm_answer_keyverify: publickey 00000025B7208450 signature verified
29708 2018-03-26 12:18:09.099 debug3: mm_request_send entering: type 25
29708 2018-03-26 12:18:09.099 debug1: monitor_child_preauth: sshd has been authenticated by privileged process
29708 2018-03-26 12:18:09.099 debug3: mm_get_keystate: Waiting for new keys
29708 2018-03-26 12:18:09.099 debug3: mm_request_receive_expect entering: type 26
29708 2018-03-26 12:18:09.099 debug3: mm_request_receive entering
29708 2018-03-26 12:18:09.099 debug3: mm_get_keystate: GOT new keys
29708 2018-03-26 12:18:09.099 debug2: userauth_pubkey: authenticated 1 pkalg ssh-rsa [preauth]
29708 2018-03-26 12:18:09.099 debug3: send packet: type 52 [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_request_send entering: type 26 [preauth]
29708 2018-03-26 12:18:09.099 debug3: mm_send_keystate: Finished sending state [preauth]
29708 2018-03-26 12:18:09.099 debug1: monitor_read_log: child log fd closed
29708 2018-03-26 12:18:09.145 debug3: spawning "C:\\Program Files\\openSSH\\sshd.exe" "-z"
29708 2018-03-26 12:18:09.145 debug3: send_rexec_state: entering fd = 5 config len 230
29708 2018-03-26 12:18:09.145 debug3: ssh_msg_send: type 0
29708 2018-03-26 12:18:09.145 debug3: send_rexec_state: done
29708 2018-03-26 12:18:09.145 debug3: ssh_msg_send: type 0
29708 2018-03-26 12:18:09.145 debug3: ssh_msg_send: type 0
29708 2018-03-26 12:18:09.145 debug3: ssh_msg_send: type 0
29708 2018-03-26 12:18:09.333 debug3: mm_request_receive entering
29708 2018-03-26 12:18:09.333 debug1: do_cleanup

**Comment by manojampalam** [comment]
Possible to share the crash dump?

**Comment by sddim** [comment]
[Report.zip](https://github.com/PowerShell/Win32-OpenSSH/files/1850968/Report.zip)

**Comment by NoMoreFood** [comment]
@sddim Can you see if there's a file for the crash located here: "C:\Users\<yourusername>\AppData\Local\CrashDumps" or here "C:\Windows\System32\Config\SystemProfile\AppData\Local\CrashDumps".  Also, if you're using password authentication, please change your password before posting the dump since it could contain sensitive information depending on where the code failed.

**Comment by sddim** [comment]
There's a crash dump:
[sshd.exe.24680.dmp.zip](https://github.com/PowerShell/Win32-OpenSSH/files/1872181/sshd.exe.24680.dmp.zip)

**Comment by NoMoreFood** [comment]
@manojampalam Looks like its failing here when attempting to dereference authctxt->pw:

```
static void
privsep_postauth(Authctxt *authctxt)
{
#ifdef DISABLE_FD_PASSING
	if (1) {
#else
	if (authctxt->pw->pw_uid == 0) {
#endif
```

@sddim Can you try with the latest release and verify the error still occurs?  Other than groups, is there anything notable about the users?

**Comment by sddim** [comment]
The only unusal thing that comes to my mind in regard to the affected user is that the user's home direcory (in C:\users\)  has a different name than the username (created by Windows, in the form user.machine, since a folder with the name of the user coincidentally already existed).

**Comment by DarwinJS** [comment]
@manojampalam - shouldn't this code look up profile folder via API ?

**Comment by manojampalam** [comment]
This is related to #1098 and should be fixed in 7.6.1.0. @sddim, please try it out and reopen if needed.
